compiler/qsc_frontend/src/resolve.rs

```diff
@@ -667,10 +667,22 @@ impl AstVisitor<'_> for With<'_> {
             }
             ast::StmtKind::Qubit(_, pat, init, block) => {
                 ast_visit::walk_qubit_init(self, init);
-                // The binding is valid after end of the statement.
-                self.resolver.bind_pat(pat, stmt.span.hi);
                 if let Some(block) = block {
-                    self.visit_block(block);
+                    self.with_scope(block.span, ScopeKind::Block, |visitor| {
```

Comment:
I think you may want to use the `with_pat` helper here. This mimics what we do for, say, for loops - this is kind of the same idea where the binding happens outside the block, but only applies to the block.

https://github.com/microsoft/qsharp/blob/b8a94e8d417dba5a8aef256103c4a14abd5937f5/compiler/qsc_frontend/src/resolve.rs#L701-L703

Reply:
Oh, great suggestion! I missed that I could use an existing function for that, definitely a better way to go.